src/main/java/org/jfrog/hudson/util/ExtractorUtils.java

```diff
@@ -580,28 +580,22 @@ private static boolean isMavenResolutionConfigured(ResolverContext resolverConte
                         resolverContext.getResolverOverrider() instanceof MavenResolver);
     }
 
-    /**
-     * The token that combines the project name and unique number to create unique workspace directory.
-     */
-    private static final String WORKSPACELIST = System.getProperty("hudson.slaves.WorkspaceList");
-
     /**
      * Create a temporary directory under a given workspace
      * @param launcher
      * @param ws
      * @throws Exception
      */
-    public static FilePath createAndGetTempDir(hudson.Launcher launcher, FilePath ws) throws Exception {
-        final String WORKSPACELIST = System.getProperty("hudson.slaves.WorkspaceList");
-        final FilePath tempDirPath = new FilePath(ws.getParent(), ws.getName() + ((WORKSPACELIST != null) ? WORKSPACELIST : "@") + "tmp");
-        launcher.getChannel().call(new MasterToSlaveCallable<Boolean, IOException>() {
-            public Boolean call() {
-                File tempDirFile = new File(tempDirPath.getRemote());
-                tempDirFile.mkdir();
+    public static FilePath createAndGetTempDir(hudson.Launcher launcher, final FilePath ws) throws Exception {
+        String workspaceList = System.getProperty("hudson.slaves.WorkspaceList"); // The token that combines the project name and unique number to create unique workspace directory.
```

Comment:
Let's put the comment above the line is describes.
It makes it a lot easier to read.